diff -up pegasus/env_var_Linux.status.orig pegasus/env_var_Linux.status
--- pegasus/env_var_Linux.status.orig	2017-02-07 10:54:55.812448005 +0100
+++ pegasus/env_var_Linux.status	2017-02-07 10:54:55.920447849 +0100
@@ -133,7 +133,7 @@ PEGASUS_LOCAL_DOMAIN_SOCKET_DIR = $(PEGA
 PEGASUS_LOCAL_DOMAIN_SOCKET_PATH = \
      $(PEGASUS_LOCAL_DOMAIN_SOCKET_DIR)/cimxml.socket
 
-PEGASUS_PEM_DIR = $(PEGASUS_CONFIG_DIR)
+PEGASUS_PEM_DIR = /etc/pki/Pegasus
 PEGASUS_SSL_KEY_FILE = file.pem
 PEGASUS_SSL_KEY_FILE_PATH = $(PEGASUS_PEM_DIR)/$(PEGASUS_SSL_KEY_FILE)
 PEGASUS_SSL_CERT_FILE = server.pem
